src/compiler/transformers/utilities.ts

```diff
@@ -273,9 +273,21 @@ namespace ts {
             case SyntaxKind.AmpersandEqualsToken: return SyntaxKind.AmpersandToken;
             case SyntaxKind.BarEqualsToken: return SyntaxKind.BarToken;
             case SyntaxKind.CaretEqualsToken: return SyntaxKind.CaretToken;
+            case SyntaxKind.BarBarEqualsToken: return SyntaxKind.BarBarToken;
+            case SyntaxKind.AmpersandAmpersandEqualsToken: return SyntaxKind.AmpersandAmpersandToken;
+            case SyntaxKind.QuestionQuestionEqualsToken: return SyntaxKind.QuestionQuestionToken;
+
         }
     }
 
+    export function shouldCaptureInTempVariable(expression: Expression): boolean {

```

Comment:
You should put this next to `isSimpleCopiableExpression` and `isSimpleInlineableExpression` as all three functions serve similar purposes with slightly different requirements. In the long term we should find a way to unify them, but for now they should all be adjacent so we can find them easily later.

This function also should probably be renamed to something more specific if its globally available to the entire compiler, otherwise someone might use it inadvertently thinking it covers more scenarios than it actually does. 

Alternatively, it might be worthwhile to see if `isSimpleInlineableExpression` might work instead of `shouldCaptureInTempVariable`.

Reply:
`isSimpleInlineableExpression` is a bit weird that it return false if expression is a identifier.

Reply:
With generators and async functions, its very possible some operation could modify the value of an identifier while your code is suspended.

Reply:
While less of a concern currently, its also possible that merely accessing an identifier could change its value if/when https://github.com/tc39/proposal-realms becomes standard, as it allows you to create a new realm for script execution where the global lexical environment can be a `Proxy` that wraps the global object.

Reply:
Thanks for the explanation.